### PR TITLE
Add to 2d software skinning classref note about custom shaders

### PR DESCRIPTION
**Pull request by lawnjelly**
Added note to say that custom shaders operate on VERTEX after the transform with 2d software skinning rather than before (as is the case with hardware skinning).

## Notes
* Given that the transform for software skinning happens on CPU it would not be trivial to make available the same functionality as hardware skinning for custom shaders. It could be done by skinning the transforms instead of the vertex and using large FVF but this may not be worth it, and would certainly require testing etc.
* We could just say reading and writing to VERTEX is unsupported for software skinning.
* On the other hand, being able to operate in world (or camera?) space is actually more useful in many situations than using pre-transformed space. So we could turn a bug into a feature... :slightly_smiling_face: 
* We have to weigh this up against the commitment to support this in future.

### CODE REVIEW

File: doc/classes/ProjectSettings.xml [hunk 1]
```diff
@@ -1046,6 +1046,7 @@
 			If [code]true[/code], performs 2D skinning on the CPU rather than the GPU. This provides greater compatibility with a wide range of hardware, and also may be faster in some circumstances.
 			Currently only available when [member rendering/batching/options/use_batching] is active.
 			[b]Note:[/b] Antialiased software skinned polys are not supported, and will be rendered without antialiasing.
+			[b]Note:[/b] Custom shaders that use the VERTEX built-in operate with VERTEX position after skinning, whereas with hardware skinning, VERTEX is the position before skinning.
```
Comment (Calinou) [inline doc/classes/ProjectSettings.xml:1049]:
```suggestion
			[b]Note:[/b] Custom shaders that use the [code]VERTEX[/code] built-in operate with [code]VERTEX[/code] position [i]after[/i] skinning, whereas with hardware skinning, [code]VERTEX[/code] is the position [i]before[/i] skinning.
```

**Review by Calinou — DISMISSED** [review]
Should be good to merge after applying the suggestion.

File: doc/classes/ProjectSettings.xml (hunk 1 above)
Comment (lawnjelly) [reply doc/classes/ProjectSettings.xml]: Done. :+1:

**Review by clayjohn — APPROVED** [review]

**Comment by akien-mga** [comment]
Thanks!

**Pull request merged**